ext/lmdb-safe/lmdb-typed.hh

```diff
@@ -665,7 +673,19 @@ public:
 
         if (sthiskey == keyString) {
           auto _id = getIDFromCombinedKey(out);
-          ids.push_back(_id.getNoStripHeader<uint32_t>());
+          uint64_t ts = LMDBLS::LSgetTimestamp(id.getNoStripHeader<string_view>());
+          uint32_t __id = _id.getNoStripHeader<uint32_t>();
+
+          if (ts < oldestts) {
+            oldestts = ts;
+            oldestid = __id;
+          }
+          ids.push_back(__id);
+        }
+
+        if (onlyOldest && ids.size() > 1) {
```

Comment:
I wonder if that test would make more sense inside the `sthiskey == keyString` block, since we only insert in this case? And then, perhaps, when `onlyOldest` is `true`, only clear/insert when `ts < oldestts`?
The end result would be the same, to be clear.

Reply:
nice, that would entirely avoid the slower path for non-LS users. Will do.